File: exercism/go/grains/grains.go
// Package grains provides functions to tally the number of grains on each square
// if you were to place one grain on the first square of a chessboard, and then
// double the number of grains on each subsequent square of the board.
package grains

import (
	"fmt"
)

// Square returns the number of grains of wheat that would be on this square.
func Square(square int) (uint64, error) {
	if square < 1 || square > 64 {
		return 0, fmt.Errorf("square out of range")
	}
	return 1 << uint(square-1), nil
}

const sumOf2to64 = 1<<64 - 1

// Total implements a sum of all the quantities on the chessboard positions.
func Total() uint64 {
	return sumOf2to64
}

File: exercism/c/grains/src/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
    if (index < 1 || index > 64) {
        return 0;
    }

    return (uint64_t) 1 << (index - 1);
}

uint64_t total(void)
{
    uint64_t t = 0;
    for (int i = 1; i <= 64; i++) {
        t += square(i);
    }

    return t;
}
